server_utils/origin.go

```diff
@@ -750,18 +754,64 @@ from S3 service URL. In this configuration, objects can be accessed at /federati
 	return originExports, nil
 }
 
+// Generate a minimally scoped auth token that allows the origin
+// to query itself for its sentinel file
+func generateSentinelCheckScitoken(resourceScope string) (string, error) {
```

Comment:
One last minor nitpick -- this function technically generates a WLCG token, which isn't the same as a scitoken. How about "generateSentinelReadToken"?